.github/workflows/build-all.yml

```diff
@@ -16,33 +21,106 @@ jobs:
    uses: ./.github/workflows/build-windows.yml
 
   collect-artifacts:
+    permissions:
+      contents: write
     runs-on: ubuntu-latest
     needs: [build-alpine, build-linux, build-linuxstatic, build-macos, build-windows]
     steps:
-      - run: echo Finished with builds
-      - name: Download workflow artifacts
-        uses: dawidd6/action-download-artifact@v2.24.3
-        with:
-          workflow: build-alpine.yml
-          path: artifacts
-      - name: Download workflow artifacts
-        uses: dawidd6/action-download-artifact@v2.24.3
-        with:
-          workflow: build-linux.yml
-          path: artifacts
-      - name: Download workflow artifacts
-        uses: dawidd6/action-download-artifact@v2.24.3
-        with:
-          workflow: build-linuxstatic.yml
-          path: artifacts
-      - name: Download workflow artifacts
-        uses: dawidd6/action-download-artifact@v2.24.3
+      - run: echo Is making new release? '${{ inputs.newRelease }}'
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: actions/download-artifact@v3
         with:
-          workflow: build-macos.yml
-          path: artifacts
-      - name: Download workflow artifacts
-        uses: dawidd6/action-download-artifact@v2.24.3
+          path: downloaded-artifacts
+
+      - name: Get previous release tag
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        id: get_previous_release
+        run: |
+          PREV_RELEASE=$(gh release list --limit 1 --exclude-drafts --exclude-pre-releases | cut -f3)
+          echo "Using release ${PREV_RELEASE}"
+          echo "prev_release=${PREV_RELEASE}" >> $GITHUB_OUTPUT
+
+      - name: Get previously released artifacts
+        if: ${{ inputs.newRelease }}
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          gh release download ${{ steps.get_previous_release.outputs.prev_release }} --dir release-artifacts
```

Comment:
Wondering if we really need this, I mean minor patch releases should be used only when something is changed on patches so the old ones are not 100% compatible with the new ones, so coping the old nodejs from the old one to the new is a bit useless? Also users are mostly using `node<major>` that will always dowload the latest so must of them will never be used at all

Reply:
It's easy enough to _remove_ binaries from a release, so I'd rather start off with a super set and trim it, rather than try and manually copy the binaries over to the new release if we need to (e.g. for 8, 10, and 12 binaries that should still work, or so we claim)

Reply:
Ok let's keep them so and we will delete them manually, agree this is easier to do